File: Sources/AlertPresenter.swift
import AppKit

/// Presents a fatal error as a modal alert, then terminates. One job, no state.
enum AlertPresenter {
    static func cameraPermissionDenied() {
        show("Camera access denied",
             "Enable camera access for GlassFace in System Settings → Privacy & Security → Camera, then relaunch.")
    }

    static func noCamera() {
        show("No camera found", "GlassFace couldn't find a usable camera on this Mac.")
    }

    private static func show(_ title: String, _ message: String) {
        let alert = NSAlert()
        alert.messageText = title
        alert.informativeText = message
        alert.addButton(withTitle: "Quit")
        alert.runModal()
        NSApp.terminate(nil)
    }
}

File: win/Sources/AlertPresenter.cs
using System.Windows;

namespace GlassFace;

/// <summary>
/// Presents a fatal error as a modal dialog, then terminates. One job, no state.
/// </summary>
internal static class AlertPresenter
{
    internal static void CameraPermissionDenied() =>
        Show("Camera access denied",
            "Enable camera access for GlassFace in Settings → Privacy & security → Camera, then relaunch.");

    internal static void NoCamera() =>
        Show("No camera found", "GlassFace couldn't find a usable camera on this PC.");

    private static void Show(string title, string message)
    {
        MessageBox.Show(message, title, MessageBoxButton.OK, MessageBoxImage.Error);
        Application.Current?.Shutdown();
    }
}
